src/Stream/src/Base/Binding/MessageConverterConfigurer.cs

```diff
@@ -0,0 +1,405 @@
+            // to MessageConverters instead of interceptors
+            if (message.Payload is byte[] && message.Headers.ContainsKey(MessageHeaders.CONTENT_TYPE))
+            {
+                return message;
+            }
+
+            // ===== 1.3 backward compatibility code part-1 ===
+            string oct = null;
+            if (message.Headers.ContainsKey(MessageHeaders.CONTENT_TYPE))
+            {
+                oct = message.Headers.Get<MimeType>(MessageHeaders.CONTENT_TYPE).ToString();
+            }
+
+            var ct = oct;
+            if (message.Payload is string)
+            {
+                if (MimeTypeUtils.APPLICATION_JSON_VALUE.Equals(oct))
+                {
+                    ct = MimeTypeUtils.APPLICATION_JSON_VALUE;
+                }
+                else
+                {
+                    ct = MimeTypeUtils.TEXT_PLAIN_VALUE;
+                }
+            }
+
+            // ===== END 1.3 backward compatibility code part-1 ===
+            if (!message.Headers.ContainsKey(MessageHeaders.CONTENT_TYPE))
+            {
+                var messageHeaders = message.Headers as MessageHeaders;
+                messageHeaders.RawHeaders[MessageHeaders.CONTENT_TYPE] = _mimeType;
+            }
+
+            IMessage result;
+            if (message.Payload is byte[])
+            {
+                result = message;
+            }
+            else
+            {
+                result = _messageConverter.ToMessage(message.Payload, message.Headers);
+            }
+
+            if (result == null)
+            {
+                throw new InvalidOperationException("Failed to convert message: '" + message + "' to outbound message.");
+            }
+
+            // ===== 1.3 backward compatibility code part-2 ===
+            if (ct != null && !ct.Equals(oct) && oct != null)
+            {
+                var messageHeaders = result.Headers as MessageHeaders;
+                messageHeaders.RawHeaders[MessageHeaders.CONTENT_TYPE] = MimeType.ToMimeType(ct);
+                messageHeaders.RawHeaders[BinderHeaders.BINDER_ORIGINAL_CONTENT_TYPE] = MimeType.ToMimeType(oct);
+            }
+
+            // ===== END 1.3 backward compatibility code part-2 ===
+            return result;
+        }
+    }
+
+    internal class InboundContentTypeEnhancingInterceptor : AbstractContentTypeInterceptor
+    {
+        public InboundContentTypeEnhancingInterceptor(string contentType)
+        : base(contentType)
+        {
+        }
+
+        public override IMessage DoPreSend(IMessage message, IMessageChannel channel)
+        {
+            var messageHeaders = message.Headers as MessageHeaders;
+            var contentType = _mimeType;
+
+            /*
+             * NOTE: The below code for BINDER_ORIGINAL_CONTENT_TYPE is to support legacy
+             * message format established in 1.x version of the framework and should/will
+             * no longer be supported in 3.x
```

Comment:
These compatibility notes could use a little more context info (like versions of _what_?)

Reply:
Comment is from the original Java streams code .. so its referencing the Java Streams code versions (i.e. the framework)  ... and since we need to interop with those versions of Java Streams I left that comment in for future reference

Reply:
I updated the comment